107-tee-boot-attestation.md

```diff
@@ -0,0 +1,360 @@
+This should have enough detail that reviewers can understand exactly what
+you're proposing, but should not include things like API designs or
+implementation.  The "Design Details" section below is for the real
+nitty-gritty.
+-->
+
+### User Stories (optional)
+
+<!--
+Detail the things that people will be able to do if this enhancement is implemented.
+Include as much detail as possible so that people can understand the "how" of
+the system.  The goal here is to make this feel real for users without getting
+bogged down.
+-->
+
+Initially, the TEE handler will work in-tandem with [SVSM], which runs in
+a tenant's firmware and provides privileged operations for CVM guests. SVSM will
+handle TEE attestation before a guest OS boots. SVSM will be responsible for
+unlocking state (provided by the keylime verifier) needed to begin the guest OS
+boot process.
+
+#### Story 1
+ * User launches a confidential VM, pointing it to the URL of the keylime
+   verifier's TEE handler to communicate with for boot attestation.
+ * CVM begins running in SVSM. SVSM gathers all necessary information needed for
+   TEE attestation. Information is dependant on the specific TEE architecture
+   the guest is running on top of (ARM CCA, Intel TDX, AMD SEV-SNP, etc...).
+ * SVSM sends all attestation information to TEE handler.
+ * TEE handler uses information to verify the boot environment.
+ * If boot attestation is successful, TEE handler reads the agent's TEE secret
+   from the registrar DB and sends to CVM. If boot attestation fails, TEE
+   handler sends error message.
+ * SVSM reads response from TEE handler. If successful, SVSM uses the TEE secret
+   to unlock some state and begin booting guest OS, with a keylime agent
+   eventually initializing and integrity monitoring continuing as normal.
+ * SVSM will extend system's TPM state to account for TEE boot attestation.
+   *This is intentially vague at the moment, as the exact design is still under
+   ongoing discussion*.
+ * Agent initializes and begins running as normal. Relevant additions are made
+   to verifier to check if TPM measurement includes the data returned from TEE
+   handler. *This is also still under ongoing discussion*.
+
+### Notes/Constraints/Caveats (optional)
+
+<!--
+What are the caveats to the proposal?
+What are some important details that didn't come across above.
+Go in to as much detail as necessary here.
+This might be a good place to talk about core concepts and how they relate.
+-->
+ * TODO
+
+### Risks and Mitigations
+
+<!--
+What are the risks of this proposal and how do we mitigate.  Think broadly.
+For example, consider both security and how this will impact the larger
+enhancement ecosystem.
+
+How will security be reviewed and by whom?
+-->
+
+The TEE handler will contain a very sensitive secret for unlocking some state
+on the CVM. This secret must be properly protected and should only be made
+available to a guest that has successfully attested.
```

Comment:
Proper owership and mode bits of the DB should be sufficient or do we need encrypted columns now (if available)?

Reply:
I don't think encrypted columns are necessary, as long as the right access permission (that is, only successfully attested guests can read a secret) are in place.